### PR TITLE
[build] Copy templates to lowercase path on Linux

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/dotnet/sdk/issues/16946

Copy our template workload pack to a lowercase destination so that it
can be detected and used when building and testing on Linux.

### CODE REVIEW

**Review by grendello — APPROVED** [review]

File: build-tools/create-packs/Directory.Build.targets [hunk 1]
```diff
@@ -126,7 +126,9 @@
         DestinationFolder="$(DotNetPreviewPath)packs\$([System.String]::Copy('%(_WLPacks.Filename)').Replace('.$(_WLPackVersion)', ''))\$(_WLPackVersion)"
     />
     <MakeDir Directories="$(DotNetPreviewPath)template-packs" />
-    <Copy SourceFiles="@(_WLTemplates)" DestinationFolder="$(DotNetPreviewPath)template-packs" />
+    <!-- Copy template packs to lowercase destination when running on Linux -->
+    <Copy Condition="'$(HostOS)' != 'Linux'" SourceFiles="@(_WLTemplates)" DestinationFolder="$(DotNetPreviewPath)template-packs" />
+    <Copy Condition="'$(HostOS)' == 'Linux'" SourceFiles="@(_WLTemplates)" DestinationFiles="@(_WLTemplates->'%(Filename)'->ToLower()->'$(DotNetPreviewPath)template-packs\%(Identity)')" />
```
Comment (jonathanpeppers) [inline build-tools/create-packs/Directory.Build.targets:131]: I think we could always do the lower case, even the .NET 5 ones are lower:

image

Comment (pjcollins) [reply build-tools/create-packs/Directory.Build.targets:131]: Sounds good, we can always copy to a lowercase location. I am also hoping to wait for a response on the issue or to chat about this quickly on Thursday to see if we should change the template pack ID / version (here and in macios) instead. If there won't be SDK changes coming that will process the packs regardless of casing then we should probably make that change instead.

Comment (jonpryor) [reply build-tools/create-packs/Directory.Build.targets:131]: Could someone explain that `DestinationFiles` expression to me?  I've never seen *multiple* `->` constructs before.  Does the rightmost `%(Identity)` apply to the "intermediate" list resulting from `@(_WLTemplates->'%(Filename)'->ToLower())`, or is it `%(_WLTemplates.Identity)`?

Comment (pjcollins) [reply build-tools/create-packs/Directory.Build.targets:131]: The `%(Identity)` in this case will be the product of the transform plus the item function, so the value of
  `@(_WLTemplates->'%(Filename)'->ToLower())` best summarizes it.

Comment (WonyoungChoi) [inline build-tools/create-packs/Directory.Build.targets:131]: It seems that the ".nupkg" extension is missing.
```suggestion
    <Copy Condition="'$(HostOS)' == 'Linux'" SourceFiles="@(_WLTemplates)" DestinationFiles="@(_WLTemplates->'%(Filename)%(Extension)'->ToLower()->'$(DotNetPreviewPath)template-packs\%(Identity)')" />
```

Comment (pjcollins) [reply build-tools/create-packs/Directory.Build.targets:131]: Yeah we do need to revisit test scope on Linux in the future, I think we still have a handful of failures to work through in even the smoke test category (#5701)

**Comment by pjcollins** [comment]
Assuming this works everywhere I think we can merge it temporarily, or we can wait for a potential resolution to https://github.com/dotnet/sdk/issues/16946.

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**